README.md

```diff
@@ -1 +1,218 @@
-Magento_UpwardConnector module used for correctly routing frontend requests through UPWARD-PHP.
+# How to install UPWARD with a PWA Studio storefront
+
+## Prerequisites
+
+- A dev, starter, or pro tier Magento Cloud project
+- Composer
+- Yarn or NPM
+- A [Magento PWA Studio][] storefront managed by NPM or Yarn.
+
+  You are not required to publish to npmjs.com, but
+  NPM or Yarn should be able to access your project code through Git.
+
+  For this tutorial, the [`venia-concept`][] package for the [Venia storefront][] is used, but any PWA available though NPM with an UPWARD compatible YAML file is supported.
+
+## Add node package dependency
+
+If it does not exist, [create a `package.json`][] file in your project directory.
+
+```sh
+npm init -y
+```
+
+Add your PWA Studio storefront to the list of dependencies in this file.
+
+```json
+{
+  "dependencies": {
+    "@magento/venia-concept": "~2.0.0"
+  }
+}
+```
+
+## Add Composer dependencies
+
+Add the repository information for the UPWARD PHP server and UPWARD Connector to the `repositories` section in your `composer.json`.
+
+```json
+"upward-connector": {
+    "type": "vcs",
+    "url": "https://github.com/magento-research/magento2-upward-connector"
+},
+"upward-php": {
+    "type": "vcs",
+    "url": "https://github.com/magento-research/upward-php"
+}
+```
+
+To have composer install these packages, add them to the `require` section of the `composer.json` file.
+
+```json
+"magento/module-upward-connector": "^1.0.0",
+"magento/upward": "^1.0.0"
```

Comment:
We'll have to circle back to this once everything is tagged, but I _think_ once we switch to a stable version of `magento/upward` within the Connector, users shouldn't need to specify it here as well.

Reply:
Maybe I'm a bit late to this, but I wonder if this should be called `upward-php-server` for clarity's sake? You know the Composer convention better than me, so if it's redundant to call it that, I get it.

Reply:
@bbatsche: Once published, that should be accurate. The docs should be usable at release, and those packages won't be published yet, so we should leave this in place. Maybe call it a `git-based` installation for development purposes?

@zetlen: I think a Composer package implies that it's PHP, and would be redundant. A name change would be much more difficult post-release, so if you want to be concise and go `upward-server`, let's make it happen now.

Reply:
Thanks for that context. I anticipate more tools for UPWARD in various ecosystems so I would like to clarify this package's purpose by calling it `upward-server`.

Reply:
https://github.com/magento-research/upward-php/pull/60